### PR TITLE
Add fpm nginx template

### PR DESCRIPTION
**Pull request by MattiasGees**

### CODE REVIEW

File: manifests/fpm.pp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  file {
+    "/etc/nginx/sites-available/${name}.conf":
+      ensure   => file,
+      content  => template('nginx/etc/nginx/sites-available/fpm.conf.erb'),
+      mode     => '0644',
+      owner    => root,
+      group    => root,
+      require  => Class['nginx::install'],
+      notify   => Class['nginx::service'];
```
Comment (iuriaranda) [inline manifests/fpm.pp]: You only need to restart nginx when creating the `sites-enabled` soft-link, right?

Comment (MattiasGees) [inline manifests/fpm.pp]: what if something changes in the file and it was already enabled?

File: templates/etc/nginx/sites-available/fpm.conf.erb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+  ssl_session_cache  shared:SSL:10m;
+  ssl_session_timeout  10m;
+  ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+  ssl_prefer_server_ciphers on;
+  ssl_ciphers AES256+EECDH:AES256+EDH:AES128+EDH:!aNULL;
+  <% end %>
+
+  root /var/www/<%= @name %>/current/public;
```
Comment (SamClinckspoor) [inline templates/etc/nginx/sites-available/fpm.conf.erb]: maybe make this the default if `root_path = undef` and use `@root_path` if it is defined.

Comment (SamClinckspoor) [inline templates/etc/nginx/sites-available/fpm.conf.erb]: or add a variable for the `/current/public` part

**Comment by SamClinckspoor** [comment]
this drupal php-fpm vhost can be some inspiration for good nginx options to set: https://www.nginx.com/resources/wiki/start/topics/recipes/drupal/

of course some are drupal specific there

**Comment by SamClinckspoor** [comment]
:shipit:

**Comment by simonrondelez** [comment]
:shipit:

**Pull request merged**